### DISCUSSION TITLE
Will DivestOS prevent me from assigning an IMEI?

### DISCUSSION THREAD

**Discussion by rsssxd** [Divested-Mobile/DivestOS-Build#41] [Q&A] [answered]
I want to set IMEI. Does it work with DivestOS? Or do I need to install another ROM?

**Comment by SkewedZeppelin** [comment, marked as answer]
Out of scope.

**Reply by rsssxd** [reply]
The economy in my country is very bad. I have to 😏

**Reply by martin-braun** [reply]
@SkewedZeppelin IMEI changing is illegal in some countries, such as call recording, yet some ROMs allow to do this, because the user breaks the law, not the dev who offers this option with a big disclaimer. 

A lot of privacy focused people claim an IMEI is a privacy violation. Even if it can disturb the mobile network, it wouldn't be a problem if the concept of IMEIs was dropped in the first place.

I honestly have to support the idea that switching an IMEI should be possible, although I would probably not consider doing this on my own, due to the global collision potential.

**Reply by SkewedZeppelin** [reply]
@martin-braun the real issue is actually doing the change.
a lot of them just change what is shown and not what is actually sent on the wire
you can change the latter via like qxdm on some older qualcomm devices easily, but not so much on newer devices afaik
mediatek devices are the only ones where you can change at runtime afaik, but I don't support mtk

I understand imei changing is quite common in some countries where you have to register your phone, and people will spoof so they don't have to pay the very costly registration fee

**Comment by Pantyhose-X** [comment]
[Although in most countries it is not illegal per se to change IMEI (or at least not prosecutable unless you use this feature for malicious purposes)
IMEI is 15 digits long - so 10^15 different numbers, while it's estimated there are less than 8*10^9 mobile phones in the world. So it's 0.000008 chance of you accidentally picking someone else's IMEI, and much less of you both being in the same network.
 And it is possible to rule out even this unlikely probability, if you change the IMEI to that of some broken/old phone which you own and/or which is guaranteed to never be used again.](https://forum.pine64.org/showthread.php?tid=14743)
